packages/rspack/src/config/normalization.ts

```diff
@@ -104,7 +104,9 @@ export const getNormalizedRspackOptions = (
 		entry:
 			config.entry === undefined
 				? { main: {} }
-				: getNormalizedEntryStatic(config.entry),
+				: getNormalizedEntryStatic(
+						typeof config.entry === "function" ? config.entry() : config.entry
+					),
```

Comment:
Allowing `config.entry` to be a function returning `EntryStatic`. 

Good enough to unlock the filename function tests, but not complete:
in webpack `config.entry` returning `Promise<EntryStatic>` is also supported.